### DISCUSSION TITLE
Google Lighthouse, performance issue:  Does not use passive listeners to improve scrolling performance

### DISCUSSION THREAD

**Discussion by tedc** [nolimits4web/swiper#4610] [Q&A]
Testing the performance of any sites that use Swiper, Lighthouse returns always this error: 
**Does not use passive listeners to improve scrolling performance**

It refers to the code below:

`for (var i = 0; i < this.length; i += 1) {
    var el = this$1[i];
    if (!targetSelector) {
      for (j = 0; j < events.length; j += 1) {
        var event = events[j];
        if (!el.dom7Listeners) { el.dom7Listeners = {}; }
        if (!el.dom7Listeners[event]) { el.dom7Listeners[event] = []; }
        el.dom7Listeners[event].push({
          listener: listener,
          proxyListener: handleEvent
        });
        el.addEventListener(event, handleEvent, capture);
      }
    }`

I can't find any workaround. Am I doing something wrong? My passiveListener option it's setted to true.

**Comment by nolimits4web** [comment]
Just ignore it. Some events are required to be "not passive"

**Reply by tedc** [reply]
Sorry but I can't ignore it, my Lighthouse it brings down my lighthouse score and this is a big SEO issue.

**Reply by tedc** [reply]
It seems a problem related to Dom7, the "capture" value has to be `{passive:true}` for touchstart and touchmove events.

**Reply by nolimits4web** [reply]
Do you know what passive listener means? If it will be passive then Swiper functionality will be broken as it won't be able to prevent touch moves, prevent it and move swiper instead. If you can't leave with this Lighthouse warning, you should probably not use Swiper then

**Reply by tedc** [reply]
Sorry but here https://github.com/nolimits4web/swiper/blob/3aad554cb4411d4d810355d701b73480f352fe71/src/components/core/events/index.js the touchstart event has passive true, so I thought that the issue was only related to the Dom7 dependencies. Swiper.js has already solved the problem. It's not a simple warning, is marked as performance issue (red, not orange or grey). Anyway, I'll leave with it, working on other details. I thought that was right reporting to you. Thanks.
